plugins/search-backend-module-techdocs/src/collators/DefaultTechDocsCollatorFactory.ts

```diff
@@ -223,16 +225,14 @@ export class DefaultTechDocsCollatorFactory implements DocumentCollatorFactory {
 
               return searchIndex.docs.map((doc: MkSearchIndexDoc) => ({
                 ...this.entityTransformer(entity),
-                title: unescape(doc.title),
-                text: unescape(doc.text || ''),
```

Comment:
Open question with no straight answer: Would it be better to keep this triplet above the transformer call, and let the transformer return a `Partial<T>` instead? That way you only override what you want automatically without calling  `defaultTechDocsCollatorDocumentTransformer` explicitly.

Mainly this helps cover cases where the return type shape changes over time, while letting caller's code not break. For example when a new field is added.

Another pattern we've explored elsewhere is to pass in an `applyDefaults` callback to the function. Then they can choose to return `{ ...applyDefaults(=, myOverride: 1 }` or whatnot. That gives the same benefits, while also not requiring a separate export to be made for the default implementation (keeps the root api a bit cleaner)

No blocker from my end. Just thinking out loud.

Reply:
Yeah, I'm tempted to agree with this thought. Although the Entity Transformer hasn't changed, this setup is still a bit brittle to changes and requires every implementation of the Transformer to return all the properties even if they're only interested in changing 1 value. Although it's not a huge type, there's still overhead (for example maybe the implementation forgets to escape the title)...

I'd opt for the transformer returning a Partial type and move the 'defaults' back into this file so it the near to where they're being used, instead of spreading that logic across multiple files.

However, I do like that this implementation follows the same pattern so it's consistent. I think we need to make the same update to the `TechDocsCollatorEntityTransformer` and deprecate the `defaultTechDocsCollatorEntityTransformer` (probably in another PR) so that we bring these two parts into alignment.

@alexef Do you have the time to make the update to this PR?

Reply:
@alexlorenzi PTAL

Reply:
Hey @alexef - This is closer, but the original idea was to keep the content of the `defaultTechDocsCollatorDocumentTransformer` in this file and completely remove that function. 

However, since we'll still need something like the `TechDocsCollatorDocumentTransformer` so that we can type the override in the parameters, I think this is an ok solution for now. I don't want to hold you up any longer. I did make some quick suggestions just to clean up a couple of parts of the code a bit, though, if. you could address those before merging.

This whole section could do with looking at again since, as you correctly point out, there are options in the Entity Transformer (and potentially) options in the Document Transformer that just aren't overridable, and that leads to a confusing API. I'll schedule some time to look over this and build a more robust solution for overriding these values, and make sure to keep you in the loop.